types/tuple.hh

```diff
@@ -123,7 +123,18 @@ public:
     const std::vector<data_type>& all_types() const {
         return _types;
     }
-    std::vector<bytes_view_opt> split(bytes_view v) const;
+    std::vector<bytes_opt> split(FragmentedView auto v) const {
+        std::vector<bytes_opt> elements;
```

Comment:
Future optimization: reserve() the tuple declared size here, on the assumption that most tuples will populate all their elements.